### ISSUE TITLE
Some commands (GIT , AZ) hang when using the dotnet interactive kernel in VS Code but are OK with the .net(powershell) on in VSCode or Jupyter.

### ISSUE THREAD

**Issue by jhoneill** [dotnet/interactive#1546]
_This may need to be logged as a VSCode problem, but because the behaviour seems to be different between two dotnet interactive kernels  I'm  starting here_

### Describe the bug
An advantage of having PowerShell as a supported environment in notebooks the ability to run any command line and capture its output. For example a powershell cell can contain
```
ping  localhost
```
 I have been been using this to create a shared notebook of useful **git** examples. 
With combination of   anaconda  / Jupyter and the  .NET (PowerShell) kernel,
```
git status
```
and similar work. Doing the same in VSCode fails with the default kernel. The command will hang indefinitely, and usually clicking the stop button returns   
`Error: System.OperationCanceledException: Command :SubmitCode: git status cancelled.` 
but I have seen this return the output (rarely).   
But I *can* switch kernels and get it to work. (See screen shots.) I'd like to use the notebooks on machines without the python / jupyter bits (only with VSCode) so this is only a partial solution. 

 It does not affect all commands  "Net share" , "ping", "Ipconfig"  "nslookup google.com", "Tasklist"  all work, so Initially I thought this was something specific to git.exe but "az -h" has the same problem 
 
I've also tried running the command as a job with `Start-Job {git status} | wait-job | receive-job` but this gives the error 
```
Start-Job: The pwsh executable cannot be found at "C:\Users\P10506111\.nuget\packages\microsoft.dotnet-interactive\1.0.240403\tools\net5.0\any\runtimes\win\lib\net5.0\pwsh.exe".
Note that 'Start-Job' is not supported by design in scenarios where PowerShell is being hosted in other applications.
```
And by trying to remote to a PSSession on my machine with    
`$pss = New-PSSession -ComputerName localhost -EnableNetworkAccess`
gives   
`New-PSSession: [localhost] An error has occurred which PowerShell cannot handle. A remote session might have ended.`

From the message the first looks to be by design, I can't tell if the second is design, bug or user error - they fail in both kernels.  I have also tried creating a PS Runspace to run the command in the default kernel - no joy with that either.  

### Please complete the following:

**Which version of .NET Interactive are you using?** There are a few ways to find this out:
Version: 1.0.240403+56a0e392fba305af06e72ce7dec885b2b6722c00
Build date: 2021-08-09T11:46:14.5291024Z

- OS
    - [X] Windows 10
 - Browser 
    - [X] Edge
 - Frontend
    - [X] Jupyter Notebook
    - [X] Visual Studio Code 
    - [X] Visual Studio Code Insiders

##Screenshots

![image](https://user-images.githubusercontent.com/16209593/129019807-3b16a622-7c40-4787-9072-206ae2a05c3c.png)

![image](https://user-images.githubusercontent.com/16209593/129020505-da63afbf-6ffa-4918-aa83-e43e74bb794a.png)

**Comment by jhoneill** [comment]
I had forgotten that I had already raised this as https://github.com/dotnet/interactive/issues/970   I'm going to close that one as this is a better description and relates to how VSCode and Dot net interactive stand now

**Comment by InTEGr8or** [comment]
I am also seeing this on `aws` and `az` commands, such as `aws --version`

**Comment by jonsequitur** [comment]
@daxian-dbw Any ideas?

**Comment by dfinke** [comment]
I set up interactive notebooks in GitHub Codespaces, `git status` worked. On my local box, `git status` hangs.

![image](https://user-images.githubusercontent.com/67258/130288980-7bfc13ac-5e7b-4708-97eb-e1d716fa21a6.png)

**Comment by daxian-dbw** [comment]
@jonsequitur No, not sure what is going on. I tried with Jupyter nootbook + latest stable version of `Microsoft.dotnet-interactive` global tool, and `git status` does work in it. What are the differences between dotnet-interactive running in VSCode and in Jupyter nootbook?

**Comment by dfinke** [comment]
@daxian-dbw it's only available in Codespaces "proper", not the github.dev version.

**Comment by jonsequitur** [comment]
This looks to me like a VS Code messaging issue. The command appears to complete once you trigger another command, e.g. running another cell or even hovering somewhere that triggers a hover text request:

![issue 1546](https://user-images.githubusercontent.com/547415/130669324-2edb368d-d40c-4e3e-8758-a6d05e86d806.gif)

**Comment by brianary** [comment]
👉 Making edits to the cell causes the command I/O to progress, either typing or deleting characters:

![dotnet-interactive-az-cli](https://user-images.githubusercontent.com/8332839/151298622-7d16ef8b-b6e6-47ed-82cf-0bd2b51204c4.gif)
